### PR TITLE
DROOLS-2296: Inline controls: Undefined Expression selector

### PR DESCRIPTION
**Pull request by manstis**
See JIRAs "Coding in progress" on https://issues.jboss.org/browse/DROOLS-2272

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/203
- https://github.com/kiegroup/kie-wb-common/pull/1453

**Comment by ederign** [comment]
@jomarko can you please take a look on this?

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/context/GridBodyCellEditContext.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.uberfire.ext.wires.core.grids.client.widget.context;
+
+import java.util.Optional;
+
+import com.ait.lienzo.client.core.shape.wires.handlers.MouseEvent;
```
Comment (jomarko) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/context/GridBodyCellEditContext.java]: This import is used just in javadoc, not sure if it is usual practice. What do you think?

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManager.java [hunk 2]
```diff
@@ -243,7 +244,8 @@ public boolean startEditingCell(final Point2D ap) {
         }
 
         return edit(uiRowIndex,
-                    ci);
+                    ci,
+                    Optional.of(ap));
```
Comment (jomarko) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManager.java]: Does the `ap` stand for absolute position? if yes, I have concern why it is passed in place of argument declared as `relativeLocation`.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java [hunk 3]
```diff
@@ -508,12 +510,22 @@ public void startEditingCellPointCoordinateOutsideGridBounds() {
...
+                             any(Callback.class));
     }
 
     @Test
     @SuppressWarnings("unchecked")
     public void startEditingCellPointCoordinateWithinGridBounds() {
         final ArgumentCaptor<GridBodyCellRenderContext> contextArgumentCaptor = ArgumentCaptor.forClass(GridBodyCellRenderContext.class);
+        final ArgumentCaptor<GridBodyCellEditContext> editContextArgumentCaptor = ArgumentCaptor.forClass(GridBodyCellEditContext.class);
 
         cellSelectionManager.startEditingCell(new Point2D(150,
```
Comment (jomarko) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java]: Could we please extract point:
```java
final Point2D editedAtPoint = new Point2D(150, 42);
```

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java [hunk 4]
```diff
@@ -522,12 +534,25 @@ public void startEditingCellPointCoordinateWithinGridBounds() {
...
                      context.getColumnIndex());
+
+        final GridBodyCellEditContext editContext = editContextArgumentCaptor.getValue();
+        assertTrue(editContext.getRelativeLocation().isPresent());
+        assertEquals(150,
+                     editContext.getRelativeLocation().get().getX(),
+                     0.0);
+        assertEquals(42,
+                     editContext.getRelativeLocation().get().getY(),
+                     0.0);
```
Comment (jomarko) [inline uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java]: And then do just one assertion:
```java
assertEquals(editedAtPoint, editContext.getRelativeLocation().get());
```

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/context/GridBodyCellEditContext.java (hunk 1 above)
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/context/GridBodyCellEditContext.java]: Well, IDEA added it for me.. and removing it gives an error in IDEA.. so prefer to keep it TBH.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManager.java (hunk 2 above)
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManager.java]: JavaDocs states ```ap``` is "Canvas coordinate relative to the GridWidget" so ```ap``` is a little misleading. I've changed it to ```rp``` throughout.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/context/GridBodyCellEditContext.java (hunk 1 above)
Comment (jomarko) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/context/GridBodyCellEditContext.java]: I am fine with it.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java (hunk 3 above)
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java]: Done.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java (hunk 4 above)
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/selections/impl/BaseCellSelectionManagerTest.java]: Done.

**Pull request merged**